Panic: Index out of range error when merging profiles

Encountered index out of range errors in the read path (ingesters):

https://github.com/grafana/pyroscope/blob/4fdd876d8144494cb483ef74089d4809021ad5a8/pkg/pprof/merge.go#L257-L261

called from:

https://github.com/grafana/pyroscope/blob/4fdd876d8144494cb483ef74089d4809021ad5a8/pkg/pprof/merge.go#L66-L68

```
golang.org/x/sync@v0.5.0/errgroup/errgroup.go:72 +0x96
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 93207350
golang.org/x/sync@v0.5.0/errgroup/errgroup.go:75 +0x56
golang.org/x/sync/errgroup.(*Group).Go.func1()
github.com/grafana/pyroscope/pkg/phlaredb/symdb/resolver.go:263 +0x138
github.com/grafana/pyroscope/pkg/phlaredb/symdb.(*Resolver).withSymbols.func1()
github.com/grafana/pyroscope/pkg/phlaredb/symdb/resolver.go:248 +0x178
github.com/grafana/pyroscope/pkg/phlaredb/symdb.(*Resolver).Pprof.func1(0x2258460?, {{0xc18379d420, 0x1, 0x1}, {0xc18379d428, 0x1, 0x1}, {0x0, 0x0, 0x0}})
github.com/grafana/pyroscope/pkg/pprof/merge.go:26
github.com/grafana/pyroscope/pkg/pprof.(*ProfileMerge).Merge(...)
github.com/grafana/pyroscope/pkg/pprof/merge.go:68 +0x62d
github.com/grafana/pyroscope/pkg/pprof.(*ProfileMerge).merge(0xc02ecaa7e0, 0xc0aee65350, 0x1?)
github.com/grafana/pyroscope/pkg/pprof/merge.go:259
github.com/grafana/pyroscope/pkg/pprof.RewriteMappings(...)
goroutine 93207317 [running]:

panic: runtime error: index out of range [0] with length 0
```

At a glance it looks like we are unable to merge profiles that have no mappings.

--- a/pkg/phlaredb/symdb/resolver_pprof_truncate.go
+++ b/pkg/phlaredb/symdb/resolver_pprof_truncate.go
@@ -262,6 +262,10 @@ func createLocationStub(profile *googlev1.Profile) {
 		SystemName: stubNodeNameIdx,
 	}
 	profile.Function = append(profile.Function, stubFn)
+	// in the case there is no mapping, we need to create one
+	if len(profile.Mapping) == 0 {
+		profile.Mapping = append(profile.Mapping, &googlev1.Mapping{Id: 1})
+	}
 	stubLoc := &googlev1.Location{
 		Id:        uint64(len(profile.Location) + 1),
 		Line:      []*googlev1.Line{{FunctionId: stubFn.Id}},

